lib/gc/Analysis/TargetDescriptionAnalysis.cpp

```diff
@@ -0,0 +1,107 @@
+//===-- TargetDescriptionAnalysis.cpp - target description impl -*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc/Analysis/TargetDescriptionAnalysis.h"
+#include <limits>
+#include <llvm/Support/Debug.h>
+#include <regex>
+
+namespace mlir {
+namespace gc {
+
+#define DEBUG_TYPE "target-description-analysis"
+
+llvm::DenseMap<DeviceType, std::string>
+    TargetDescriptionAnalysisBase::DeviceKeyMap = {
+        {CPU, "CPU"},
+};
+
+// default values for properties
+llvm::DenseMap<StringRef, int64_t>
+    CPUTargetDescriptionAnalysis::CPUTargetDeafultValueMap = {
```

Comment:
`uint64`?

Reply:
Besides, is it single `int` type enough to describe CPU target? i.e. some CPU flag, such as `is_amx` or `is_vnni` seems better to use `bool`  to represent. How do we express all of them?

Reply:
Good Catch! I changed the data type to `unsigned` following [llvm target info](https://github.com/llvm/llvm-project/blob/5c56b46a32a8856a022a54291bc9294068f7ddbd/llvm/lib/Analysis/TargetTransformInfo.cpp#L795). The default value is decided by each `getXXX` function so that they can control their own data type.